src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../contexts/TodosContext';
+
+interface Props {
+  todo: Todo,
+  isTempTodo?: boolean,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo: {
+    id,
+    title,
+    completed,
+  },
+  isTempTodo = false,
+}) => {
+  const { deleteTodo } = useContext(TodosContext);
+  const [isDeliting, setIsDeliting] = useState(false);
+  const [isEditing] = useState(false);
+
+  const handleDeleteButton = () => {
+    setIsDeliting(true);
+
+    if (deleteTodo) {
+      deleteTodo(id)
+        .finally(() => {
+          setIsDeliting(false);
+        });
+    }
+  };
+
+  return (
+    <div data-cy="Todo" className={`todo ${completed && 'completed'}`}>
```

Comment:
better to use classnames package for such cases to avoid it
<img width="347" alt="image" src="https://github.com/mate-academy/react_todo-app-add-and-delete/assets/70220667/659cce0b-ab02-4558-833d-69a5f2d429ca">